src/components/PostDetails.tsx

```diff
@@ -1,117 +1,160 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import * as commentService from '../services/comments';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  selectedPost: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [writeComment, setWriteComment] = useState(false);
+
+  function loadComments() {
+    setErrorMessage('');
+    setWriteComment(false);
+    setLoading(true);
+
+    if (!selectedPost.id) {
+      setWriteComment(false);
+
+      return;
+    }
+
+    commentService.getComments(selectedPost.id)
+      .then(setComments)
+      .catch(() => setErrorMessage('Something went wrong!'))
+      .finally(() => setLoading(false));
+  }
+
+  useEffect(loadComments, [selectedPost.id]);
+
+  const addComment = ({
+    postId,
+    name,
+    email,
+    body,
+  }: Comment) => {
+    return commentService.createComment({
+      postId,
+      name,
+      email,
+      body,
+    })
+      .then(newComment => {
+        setComments(currentComments => [...currentComments, newComment]);
+      })
+      .catch(() => {
+        setWriteComment(false);
+        setErrorMessage('Something went wrong!');
+      });
+  };
+
+  const deleteComment = (commentId: number) => {
+    setComments(currentComments => currentComments
+      .filter(comment => comment.id !== commentId));
+
+    return commentService.deleteComment(commentId);
+  };
+
+  const hasError = !loading && errorMessage;
+  const hasNoComments = !loading
+    && !errorMessage && selectedPost && !comments.length;
+  const shouldRenderComments = !loading
+    && !errorMessage && selectedPost && comments.length > 0;
```

Comment:
Do not repeat your self, you already have it [here](https://github.com/mate-academy/react_dynamic-list-of-posts/pull/1029/files#diff-26ad4b834941d9b19ebf9db8082bd202aaf72ea0ddea85f5a8a0cb3c729cc6f2R46)